### PR TITLE
Rewrite of contribution section of Help and Support page.

### LINKED ISSUE #746: Update help&support page
### Description
It [suggests](https://docs.conda.io/en/latest/help-support.html#contribute-recipes-to-github) users can help by contributing recipes to https://github.com/conda/conda-recipes, but that repo is archived now. If anything, it should point to conda-forge?

### Tasks
- [x] Should this point to github.com/AnacondaRecipes/ (If not, where?) - AnacondaRecipes is a good place to get feedstocks, but not to contribute back to. Will link to AnacondaRecipes, conda-forge, and bioconda.
- [ ] Recommend a way this should work (the docs landing page will likely be replaced by conda.org work)
- [ ] Fix it if it's simple

### PR DESCRIPTION
**Pull request by kathatherine**
### Description

Closes #746. Added information on AnacondaRecipes, but clarified that you can only get feedstock repos from it, not contribute to it, so I added conda-forge and bioconda as options for submission. The conda-forge link has pretty immediate docs in the README about how to contribute, but I don't know if we want to be more explicit about how to do so. Might be outside the scope of this issue.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: docs/source/help-support.rst [hunk 1]
```diff
@@ -32,16 +32,23 @@ request new features and submit any other comments you may have.
...
-https://github.com/conda/conda-recipes. When you build a new conda
-recipe for a program not yet listed in the repository, use a GitHub
-pull request to submit it to the repository, so that others using
-that program in the future can reuse it. Recipes are welcome for
-programs that use any license, such as GPL, BSD, MIT or Apache, and
-all of the recipes in thisrepository are released into the public
-domain.
+============================
+
+The conda organization has transitioned into using feedstocks, which
```
Comment (kenodegard) [inline docs/source/help-support.rst:37]:
```suggestion
The conda community has transitioned into using feedstocks, which 
```

File: news/746-update-helpsupport-page [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* (#746) Updated the Help and Support page to include more up to date information on feedstocks, where to access existing feedstocks, and where to contribute new feedstocks.
```
Comment (kenodegard) [inline news/746-update-helpsupport-page:15]: Almost! Let's move the PR# to the end.

```suggestion
* Updated the Help and Support page to include more up-to-date information on feedstocks, where to access existing feedstocks, and where to contribute new feedstocks. (#823)
```

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: news/746-update-helpsupport-page (hunk 2 above)
Comment (kathatherine) [reply news/746-update-helpsupport-page:15]: Wait, is it supposed to be the PR number and not the issue number? @kenodegard

Comment (kenodegard) [reply news/746-update-helpsupport-page:15]: I use PR # as an issue # can be nonexistent or could refer to multiple PRs. A PR # will always exist and be unique for a given change.

Comment (kathatherine) [reply news/746-update-helpsupport-page:15]: Gotcha, good to know!

**Review by kenodegard — DISMISSED** [review]

File: docs/source/help-support.rst [hunk 3]
```diff
@@ -32,16 +32,23 @@ request new features and submit any other comments you may have.
...
+The conda community has transitioned into using feedstocks, which 
+are repositories that contain package recipes and all of the necessary 
+configurations for building those recipes. This enables these packages
+to be automatically built using continuous integration (CI) services.
+
+You can clone or fork many package feedstocks from `Anaconda Recipes
+<https://github.com/AnacondaRecipes>`_, though you can't submit new
+feedstocks to that GitHub organization. To contribute new conda package
+feedstocks, submit them to `conda-forge 
+<https://github.com/conda-forge/staged-recipes>`_` or `bioconda
```
Comment (kenodegard) [inline docs/source/help-support.rst:46]:
```suggestion
<https://github.com/conda-forge/staged-recipes>`_ or `bioconda 
```

**Review by kenodegard — APPROVED** [review]

**Pull request merged**